backend/test_scoring_feature.py

```
@@ -0,0 +1,267 @@
+
+
+# ============================================================================
+# EDGE CASES
+# ============================================================================
+
+class TestEdgeCases:
+    """Test boundary conditions."""
+    
+    def test_empty_list(self):
+        """Empty list returns empty."""
+        assert score_and_rank_questions([], 0, false) == []
```

Ruff (0.15.6)

[error] Undefined name `false`

(F821)